src/Plugin/pathauto/AliasType/EntityAliasTypeBase.php

```diff
@@ -319,4 +315,39 @@ protected function isContentTranslationEnabled($bundle) {
     return $this->moduleHandler->moduleExists('content_translation') && \Drupal::service('content_translation.manager')->isEnabled($this->getEntityTypeId(), $bundle);
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function applies($object) {
+    $definition = $this->entityManager->getDefinition($this->getDerivativeId());
+    $class = $definition->getClass();
+    return ($object instanceof $class);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getPatternDescription() {
+    return $this->t('Replace this description with proper annotation effort.');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getSourcePrefix() {
+    if (empty($this->prefix)) {
+      $entity_type = $this->entityManager->getDefinition($this->getDerivativeId());
+      $path = $entity_type->getLinkTemplate('canonical');
+      // Remove slug(s)... This could probably be done cleaner, but I'm not in the mood.
+      $path_parts = explode('/', $path);
```

Comment:
This is the *prefix* so we don't support something like some/path/{slug}/suffix anyway. Given that, I think you can simplify this whole thing to

// The prefix is the path of the part until the first slug, also cut of the / before that.
$this->prefix = substr($path, 0, strpos($path, '{') - 1);